### PR TITLE
feat: Display user link in self user profile SQSERVICES-1138

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1138" title="SQSERVICES-1138" target="_blank">StorySQSERVICES-1138</a>  [iOS] Display user link in self user profile
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Display user link in self user profile.

### Attachments (Optional)

Screenshot 2022-01-18 at 16 31 21

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/Helpers/URL+Wire.swift [hunk 1]
```diff
@@ -96,6 +97,13 @@ extension BackendEnvironment {
         return shared.teamsURL.appendingPathComponent(path)
     }
 
+    fileprivate static var selfUserProfileLink: URL? {
```
Comment (David-Henner) [inline Wire-iOS/Sources/Helpers/URL+Wire.swift:100]: I would keep the name consistent between `selfUserProfileLink` and `selfUserProfile` below, what do you think?

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift [hunk 2]
```diff
@@ -261,6 +267,21 @@ extension SettingsCellDescriptorFactory {
         return textValueCellDescriptor(propertyName: .domain, enabled: false)
     }
 
+    func profileLinkTitle() -> SettingsCellDescriptorType {
+        typealias Settings = L10n.Localizable.Self.Settings
+        typealias Account = Settings.AccountSection
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift:272]: You could just do `typealias Account = L10n.Localizable.Self.Settings.AccountSection`, right?

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCopyButtonCellDescriptor.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            delegate = iconActionCell
+            iconActionCell.configure(with: copyInProgress ? copiedLink : copyLink,
+                                     variant: .dark)
+        }
+    }
+
+    // MARK: - Helpers
+
+    typealias GuestRoom = L10n.Localizable.GuestRoom
+    typealias Actions = GuestRoom.Actions
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCopyButtonCellDescriptor.swift:49]:
```suggestion
    typealias Actions =  L10n.Localizable.GuestRoom.Actions
```

Comment (David-Henner) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCopyButtonCellDescriptor.swift:49]: I think `GuestRoom` isn't used anywhere else

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift [hunk 4]
```diff
@@ -480,3 +480,40 @@ final class SettingsStaticTextTableCell: SettingsTableCell {
...
+    }
+
+    private func createConstraints() {
+        [label].prepareForLayout()
+        NSLayoutConstraint.activate([
+            label.topAnchor.constraint(equalTo: contentView.topAnchor),
+            label.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            label.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+            label.bottomAnchor.constraint(equalTo: contentView.bottomAnchor)
+        ])
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:516]: I think you could do this in one line with the `fitIn(view:insets:)` method

**Review by David-Henner — COMMENTED** [review]
In the snapshots, the link font color is different from the screenshot in your PR description, is this expected?

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift (hunk 4 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift]: Done

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift (hunk 2 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: Yes. Fixed

File: Wire-iOS/Sources/Helpers/URL+Wire.swift (hunk 1 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/Helpers/URL+Wire.swift:100]: Yes, it makes sense. 
Done

**Comment by KaterinaWire** [comment]
> In the snapshots, the link font color is different from the screenshot in your PR description, is this expected?

No. I fixed it

**Review by David-Henner — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]

**Pull request merged**